src/app/components/core/control-dispatcher/api-flow/api-flow.component.ts

```diff
@@ -0,0 +1,248 @@
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+    }
+
+    getAndExecuteAPIFlow(name) {
+        /** request started */
+        this.ngProgress.start();
+        this.controlAPIFlowService.getAPIFlow(name)
+          .subscribe((af: APIFlow) => {
+            this.ngProgress.done();
+            let v = <FormArray>this.apiFlowForm.controls['variables'];
+            v.clear();
+            this.fillParameters(af.variables);
+            // TODO: FOGL-8079 (blank values for variables are not allowed)
+            // REMOVE:
+            // af.variables = {}
+            // ^ for forced testing only
+
+            if(Object.entries(af.variables).length > 0) {
+              this.openModal('confirmation-execute-dialog', af)
+            } else {
+              this.requestAPIFlow(af.name, {});
+            }
+          }, error => {
+            /** request completed */
+            this.ngProgress.done();
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          });
+    }
+
+    deleteAPIFlow(id: string) {
+      this.controlAPIFlowService.deleteAPIFlow(this.epName) 
+      .subscribe(
+        (data: any) => {
+          /** request completed */
+          this.ngProgress.done();
+
+          // Remove from local arrays of apiFlows
+          this.apiFlows = this.apiFlows.filter(api => api.name !== this.epName);  
+
+          this.alertService.success(data.message, true);
+          this.closeModal(id);
+        },
+        error => {
+          /** request completed but error */
+          this.ngProgress.done();
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+    }
+
+    requestAPIFlow(name, payload) {
+      this.controlUtilsService.requestAPIFlow(name, payload);
+    }
+
+    getUsers() {
+      this.ngProgress.start();
+      this.userService.getAllUsers()
+        .subscribe(
+          (userData) => {
+            /** request completed */
+            this.ngProgress.done();
+            this.allUsers = userData['users'].map(user => {
+              return user.userName;
+            });
+          },
+          error => {
+            /** request completed */
+            this.ngProgress.done();
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+        });
+    }
+
+    setEdit(name, state){
+      this.editMode["name"] = name;
+      this.editMode["edit"] = state;
+      if (state == false){
+        this.updateDescription();
+      }
+    }
+
+    resetEditMode() {
+      this.editMode = {name: null, edit: false, newVal: null}
```

Comment:
newVal should be _value_, we are setting _value_ in  descriptionChange and using that in updateDescription